File: In C#/romantoint.cs
using System;
using System.Collections.Generic;

class Roman
{
    public static int RomanToInt(string s)
    {
        var roman = new Dictionary<char, int> {
            {'I', 1}, {'V', 5}, {'X', 10},
            {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}
        };

        int total = 0, prev = 0;
        for (int i = s.Length - 1; i >= 0; i--)
        {
            int val = roman[s[i]];
            if (val < prev)
                total -= val;
            else {
                total += val;
                prev = val;
            }
        }
        return total;
    }

    static void Main()
    {
        string s = "MCMXCIV";
        Console.WriteLine($"Roman numeral: {s} → Integer: {RomanToInt(s)}");
    }
}

File: In Rust/romantoint.rs
use std::collections::HashMap;

fn roman_to_int(s: &str) -> i32 {
    let mut roman = HashMap::new();
    roman.insert('I', 1); roman.insert('V', 5); roman.insert('X', 10);
    roman.insert('L', 50); roman.insert('C', 100); roman.insert('D', 500); roman.insert('M', 1000);

    let mut total = 0;
    let mut prev = 0;
    for c in s.chars().rev() {
        let val = roman[&c];
        if val < prev {
            total -= val;
        } else {
            total += val;
            prev = val;
        }
    }
    total
}

fn main() {
    let s = "MCMXCIV";
    println!("Roman numeral: {} → Integer: {}", s, roman_to_int(s));
}
